clients/src/main/java/org/apache/kafka/clients/tools/VerifiableProducer.java

```diff
@@ -150,8 +174,24 @@ public static VerifiableProducer createFromArgs(String[] args) {
             producerProps.put(ProducerConfig.ACKS_CONFIG, Integer.toString(res.getInt("acks")));
             // No producer retries
             producerProps.put("retries", "0");
+            
+            // Properties specified using the --producer-prop key=val format override
+            // other properties
+            List<String> userSpecifiedProducerProps = res.getList("userSpecifiedProducerProps");
+            for (String prop: userSpecifiedProducerProps) {
```

Comment:
Right, I just thought a few tools allowed a set of generic config opts in opt=value form already, but maybe those are all housed in core. Maybe we need a few more tools here before we can figure out which things should be refactored into utilities.